FRZDatabaseViewMapper.m

```diff
@@ -378,6 +369,55 @@ - (instancetype)init
     return self;
 }
 
+- (void)insertIndexPathForItem:(NSUInteger)item inSection:(NSUInteger)section
+{
+    if (![self.insertedSections containsIndex:section]) {
+        [self.insertedIndexPaths addObject:[NSIndexPath indexPathForItem:item inSection:section]];
+    }
+}
+
+- (void)updateIndexPathForItem:(NSUInteger)item inSection:(NSUInteger)section
+{
+    if (![self.deletedSections containsIndex:section]) {
+        [self.updatedIndexPaths addObject:[NSIndexPath indexPathForItem:item inSection:section]];
+    }
+}
+
+- (void)deleteIndexPathForItem:(NSUInteger)item inSection:(NSUInteger)section
+{
+    if (![self.deletedSections containsIndex:section]) {
+        [self.deletedIndexPaths addObject:[NSIndexPath indexPathForItem:item inSection:section]];
+    }
+}
+
+- (void)moveIndexPathForItem:(NSUInteger)oldItem inSection:(NSUInteger)oldSection
+          toIndexPathForItem:(NSUInteger)newItem inSection:(NSUInteger)newSection
+{
+    NSIndexPath *oldIndexPath = [NSIndexPath indexPathForItem:oldItem inSection:oldSection];
+    NSIndexPath *newIndexPath = [NSIndexPath indexPathForItem:newItem inSection:newSection];
+    
+    if ([self.insertedIndexPaths containsObject:newIndexPath]) {
```

Comment:
Should this maybe also check if the old indexPath was deleted?

Reply:
True, better safe than sorry :)